### PR TITLE
CIV-7322 Vary Judgment applications must be with notice

### PR DESCRIPTION
**Pull request by pliao-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-7322

### Change description ###

Vary Judgment applications must be with notice

test branch at
https://github.com/hmcts/civil-general-apps-ccd-definition/pull/600

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/api/steps.js [hunk 1]
```diff
@@ -37,6 +38,8 @@ const data = {
     '10800','FEE0443'),
   INITIATE_GENERAL_APPLICATION_NO_STRIKEOUT: genAppData.gaTypeWithNoStrikeOut(),
   INITIATE_GENERAL_APPLICATION_STAY_CLAIM: genAppData.gaTypeWithStayClaim(),
+  INITIATE_GENERAL_APPLICATION_VARY_JUDGEMENT: (iwWithNotice, generalAppN245FormUpload) => genAppData.createGAData(iwWithNotice,null,
```
Comment (karthick-mohanasundaram-HMCTS) [inline e2e/api/steps.js:41]: a small typo in iwWithNotice

File: e2e/steps_file.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ const confirmDetailsPage = require('./fragments/confirmDetails.page');
 
 const applicationTypePage = require('./pages/generalApplication/applicationType.page');
 const hearingDatePage = require('./pages/generalApplication/hearingDate.page');
-const n245FormPage = require('./pages/generalApplication/n245Form.page');
+//const n245FormPage = require('./pages/generalApplication/n245Form.page');
```
Comment (karthick-mohanasundaram-HMCTS) [inline e2e/steps_file.js:57]: is this intentional? if so, May I know why we are commenting it out?

File: e2e/api/steps.js (hunk 1 above)
Comment (pliao-hmcts) [reply e2e/api/steps.js]: fixed

File: e2e/steps_file.js (hunk 2 above)
Comment (pliao-hmcts) [reply e2e/steps_file.js:57]: this is requirement, applicant don't need to upload n245

File: e2e/pages/generalApplication/applicationTab.page.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ module.exports = {
       await I.wait(3);
     }
     await I.waitInUrl('#Applications', 10);
-    await I.waitForText('Application type', 10, locate('td[id*="GaAppDetails"] th span').first());
+    await I.waitForText('Application type');
```
Comment (madhan-mahadevan) [inline e2e/pages/generalApplication/applicationTab.page.js:13]: why did you reverted back my changes @pliao-hmcts ?

File: e2e/pages/generalApplication/checkYourAnswers.page.js [hunk 4]
```diff
@@ -5,24 +5,27 @@ module.exports = {
   fields: {
     checkAnswerForm: {
       classname: '.check-your-answers',
-      changeLink: '.case-field-change a',
-      uploadEvidenceLink: '.collection-field-table a'
```
Comment (madhan-mahadevan) [inline e2e/pages/generalApplication/checkYourAnswers.page.js:9]: why did you reverted back my changes @pliao-hmcts

File: e2e/pages/generalApplication/checkYourAnswers.page.js [hunk 5]
```diff
@@ -5,24 +5,27 @@ module.exports = {
...
     I.seeInCurrentUrl(caseId);
     I.seeNumberOfVisibleElements('.button', 2);
     if ('yes' === consentCheck) {
-      I.seeNumberOfVisibleElements(this.fields.checkAnswerForm.changeLink, 9);
+      I.seeNumberOfVisibleElements('.case-field-change a', 9);
     } else if ('no' === consentCheck) {
-      I.seeNumberOfVisibleElements(this.fields.checkAnswerForm.changeLink, 10);
+      if ('yes' === vary) {
+        I.seeNumberOfVisibleElements('.case-field-change a', 9);
+      } else {
```
Comment (madhan-mahadevan) [inline e2e/pages/generalApplication/checkYourAnswers.page.js:21]: why did you reverted back my changes @pliao-hmcts

**Review by madhan-mahadevan — CHANGES_REQUESTED** [review]

File: e2e/pages/generalApplication/checkYourAnswers.page.js (hunk 4 above)
Comment (pliao-hmcts) [reply e2e/pages/generalApplication/checkYourAnswers.page.js:9]: because it doesn't work on my local

**Pull request closed without merging**